### ISSUE TITLE
False-positive E0117 when file is included in module tree multiple times

### ISSUE THREAD

**Issue by ondt** [intellij-rust/intellij-rust#7966]
## Environment
* **IntelliJ Rust plugin version:** 0.4.157.4174-212
* **Rust toolchain version:** rustc 1.55.0 (c8dfcfe04 2021-09-06)
* **IDE name and version:** CLion 2021.2.1
* **Operating system:** NixOS 21.05 (Linux 5.13.13-zen1)

## Problem description
I'm getting `Only traits defined in the current crate can be implemented for arbitrary types [E0117]` error in an unrelated file when using the `include!()` macro in `build.rs`. 

It compiles fine using `cargo run`.

![image](https://user-images.githubusercontent.com/20520951/137209247-3cf5bf2e-6a84-4697-9b18-6e0a25107bc0.png)

## Steps to reproduce
```rust
// src/main.rs

mod foo;
use crate::foo::FooStruct;

impl PartialEq for FooStruct {
    fn eq(&self, other: &Self) -> bool {
        self.field == other.field
    }
}

fn main() {
    println!("Hello, world!");
}
```

```rust
// src/foo.rs

pub struct FooStruct {
    pub field: i32,
}
```

```rust
// build.rs

include!("src/foo.rs");

fn main() {
    let _a = FooStruct { field: 123 };
}
```

```toml
# Cargo.toml

[package]
name = "bug-report"
version = "0.1.0"
edition = "2018"
build = "build.rs"
```

**Comment by ondt** [comment]
Sometimes I get this error even if I replace `include!("src/foo.rs");` with `#[path = "src/foo.rs"] mod foo;`, but I can't create a minimal reproducible example. It just pops up randomly in a larger project I'm working on.

```rust
// build.rs

#[path = "src/foo.rs"]
mod foo;
use foo::FooStruct;

fn main() {
    let _a = FooStruct { field: 123 };
}
```
